File: EthCoffee.api/Helpers/FilterParams.cs
using System;

namespace EthCoffee.api.Helpers
{
    public class FilterParams
    {
        public string Category { get; set; } = "";      
        public string Title { get; set; } = "";
        public bool WatchlistOnly { get; set; } = false;
        public bool MyListingsOnly { get; set; } = false;
        public DateTime DateAdded { get; set; }
    }
}

File: EthCoffee-SPA/src/app/_models/filter-params.ts
import { ListedDates, SortBy } from './listing';

export interface FilterParams {
  dateAdded: ListedDates;
  category: string;
  title: string;
  sortBy: SortBy;
}

export interface UserListsParams {
  myListingsOnly: boolean;
  watchlistOnly: boolean;
}
